source/dynamical_systems/README.md

```diff
@@ -208,4 +217,114 @@ while (...) {
+- **planar_gain**; the proportional gain acting in the local plane towards the limit cycle radius
+- **normal_gain**; the proportional gain acting towards the local plane
+- **circular_velocity**; the expected angular orbital velocity around the local origin
+
+Each parameter has corresponding `set_` and `get_` functions.
+
+The constructor takes additional optional arguments to define the gain and circular velocity.
+The scalar value for the gain sets both the planar and normal gain of the DS. 
+
+Note that the argument for the radius is only present when a CartesianState is provided,
+and not when an Ellipsoid is provided.
+```c++
+// construct the circular DS with optional parameters (default values are shown)
+double radius = 1.0;
+double gain = 1.0;
+double circular_velocity = M_PI / 2;
+
+// construction with a CartesianState center
+DynamicalSystems::Circular circularDS(center, radius, gain, circular_velocity);
+
+// construction with an Ellipsoid ellipse
+DynamicalSystems::Circular circularDS(ellipse, gain, circular_velocity);
+```
+
+In addition, the following helper are defined to configure the limit cycle behaviour.
+
+```c++
+.set_center(CartesianState("center"));  // sets the center position and orientation of the limit cycle
+.set_gain(1.0);                         // sets both planar and normal gain to the gain value
+.set_rotation_angle(0.0);               // sets a rotation offset around the local Z axis
+.set_radius(1.0);                       // sets both elliptical axes to the given radius (forces circle)
+.set_radiuses({1.0, 2.0});              // sets the length of each elliptical axis individually  
+```
+
+
+## Ring
+
+The **Ring** DS is similar to the **Circular** DS but is more parameterizable.
+In brief, the strength of the limit cycle can be configured with a width around the ring radius.
+
+It only supports the `CartesianState` type, and always acts in a circular ring. 
+
+```c++
+state_representation::CartesianState center("center");
+DynamicalSystems::Ring ringDS(center);
+```
+
+### Configuring the Ring DS
+
+The Ring DS has the following core parameters:
+
+- **center**; the ring center CartesianPose expressed in the DS base reference frame.
+This sets both the origin center and the inclination of the ring plane.
+- **rotation_offset**; the orientation offset of the orientational attractor in the ring frame. 
+- **radius**; the ring radius. [m]
+- **width**; the distance from the radius where the velocity has tangential components around the ring.
+Beyond this width, the velocity is always perpendicular towards the radius. [m]
+- **speed**; the desired linear speed when travelling along the circle radius. [m/s]
+- **field_strength**; the scale factor applied to the ring speed outside of the radius + width zone.
+- **normal_gain**; the scale factor for the speed normal to the ring plane.
+- **angular_gain**; the scale factor for angular velocity restitution.
```

Comment:
Speed is for the linear restitution. Angular speed is a bit trickier to define independently since it depends on the angular error as well as the "orbital velocity" around the centre. 

I think instead of "the scale factor for angular velocity restitution" I meant to write "the scale factor for angular error restitution"

And thanks, I honestly didn't know that for sure - I will add a line to say the Circular only acts in linear space